build/patches/add-support-for-prefs-addon.patch

```diff
@@ -0,0 +1,109 @@
++++ b/chrome/android/java/src/org/chromium/chrome/browser/settings/MainSettings.java
+@@ -112,6 +112,8 @@ public class MainSettings extends PreferenceFragmentCompat
+     private void createPreferences() {
+         SettingsUtils.addPreferencesFromResource(this, R.xml.main_preferences);
+ 
++        createBromitePreferences();
++
+         cachePreferences();
+ 
+         updatePasswordsPreference();
+@@ -257,4 +259,9 @@ public class MainSettings extends PreferenceFragmentCompat
+             }
+         };
+     }
++
++    private void createBromitePreferences() {
++        SettingsUtils.inflatePrefsAddon(this, "main_preferences_addon_",
++            R.xml.class.getFields());
++    }
+ }
+diff --git a/chrome/android/proguard/main.flags b/chrome/android/proguard/main.flags
+--- a/chrome/android/proguard/main.flags
++++ b/chrome/android/proguard/main.flags
+@@ -63,3 +63,10 @@
+ -keep,allowobfuscation public class ** extends org.chromium.chrome.browser.base.SplitCompat*$Impl {
+   public <init>();
+ }
++
++# Keep fields name for all R.xml
++# used in SettingsUtils.java
++# components/browser_ui/settings/android/java/src/org/chromium/components/browser_ui/settings
++-keepclassmembers class **.R$xml* {
++    public static <fields>;
++}
+diff --git a/components/browser_ui/settings/android/java/src/org/chromium/components/browser_ui/settings/SettingsUtils.java b/components/browser_ui/settings/android/java/src/org/chromium/components/browser_ui/settings/SettingsUtils.java
+--- a/components/browser_ui/settings/android/java/src/org/chromium/components/browser_ui/settings/SettingsUtils.java
++++ b/components/browser_ui/settings/android/java/src/org/chromium/components/browser_ui/settings/SettingsUtils.java
+@@ -24,6 +24,12 @@ import androidx.preference.PreferenceFragmentCompat;
+ 
+ import org.chromium.base.ApiCompatibilityUtils;
+ 
++import org.chromium.base.Log;
++import androidx.preference.PreferenceScreen;
++import androidx.preference.PreferenceManager;
++import android.content.res.Resources;
++import java.lang.reflect.Field;
++
+ /**
+  * A helper class for Settings.
+  */
+@@ -126,4 +132,43 @@ public class SettingsUtils {
+         ImageView imageButton = (ImageView) button;
+         return imageButton.getDrawable() == parentMenu.getOverflowIcon();
+     }
++
++    public static PreferenceScreen inflatePrefsAddon(PreferenceFragmentCompat preferenceFragment,
++                                                     String name,
++                                                     Field[] fields) {
++        Resources resources = preferenceFragment.getResources();
++
++        for(int count=0; count < fields.length; count++) {
++            String assetName = fields[count].getName();
++            if (assetName.startsWith(name)) {
++                int resId = 0;
++                try {
++                    resId = fields[count].getInt(null);
++                } catch (IllegalAccessException e) {}
```

Comment:
I see this is still here; I am sorry, this will not be merged.

I am currently testing the patch but I cannot merge it with this.

Reply:
sorry i missed it. I fix it

Reply:
fixed in https://github.com/bromite/bromite/pull/857/commits/e65cfb40ec64cba44d61323d3a35a0b02178bd95